### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by carusogabriel**
Simple version of #6853, just using `assertNull`, `assertFileExists`, `assertTrue` and `assertFalse`. Also, in `develop` branch.

### CODE REVIEW

**Review by lcobucci — COMMENTED** [review]
Awesome, LGTM 👍

**Review by lcobucci — APPROVED** [review]

**Comment by lcobucci** [comment]
Since @guilhermeblanco is rebasing `develop` I'll leve this to him.

**Comment by carusogabriel** [comment]
I've improved more tests:
- Instead of `count` PHP method, use [`assertCount`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertCount);
- Avoid use PHP functions `is_*`. Instead, use [`assertInternalType`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertInternalType);
- Use [`assertEquals`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertEquals) instead of comparisions `==` and [`assertSame`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertSame) instead of strict comparasions `===`;
- Use [`assertStringEqualsFile`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertStringEqualsFile) when comparing strings and files;
- Use [`assertTrue`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertTrue) or [`assertFalse`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertFalse), instead of comparing `true` or `false` keyworkds with [`assertEquals`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertEquals)/[`assertSame`](https://phpunit.de/manual/current/pt_br/appendixes.assertions.html#appendixes.assertions.assertSame);

File: docs/en/reference/batch-processing.rst [hunk 1]
```diff
@@ -174,7 +174,7 @@ problems using the following approach:
         // do stuff with the data in the row, $row[0] is always the object
     
         // detach all entities from Doctrine, so that Garbage-Collection can kick in immediately
-        $this->em->clear();
+        $this->_em->clear();
```
Comment (Majkl578) [inline docs/en/reference/batch-processing.rst]: This change is incorrect.

**Comment by carusogabriel** [comment]
I’m trying to rebase, but I guess I lost my work 😭

**Comment by Majkl578** [comment]
You've probably just incorrectly rebased, but your commits still exist: 6999d39 + 60cd9cf.
Hint: Sometimes it's easier to just hard-reset local copy and re-apply the commits again by cherry-picking them.

**Comment by carusogabriel** [comment]
I’m gonna try this, thanks for the tip

**Comment by carusogabriel** [comment]
Should we refactory `assertEquals('', $foo)/assertSame('', $foo)` with `assertEmpty($foo)`, and `assertEquals([], $bar)/assertSame([], $bar)` with `assertEmpty($bar)`?

**Comment by Ocramius** [comment]
Don't think so: `assertEmpty()` is too lax

**Review by Majkl578 — APPROVED** [review]
Develop has already been rebased so this can be merged now.

**Comment by Majkl578** [comment]
@carusogabriel Thanks!

**Comment by carusogabriel** [comment]
Any time @Majkl578 :man_technologist:

**Comment by carnage** [comment]
Took a quick look at back porting this one and well, lets just say it's potentially going to be a big job. 

94 out of 96 files have conflicts, a 95th is a deleted/modified conflict.

One of the biggest things I've noticed is there are conflicts on a lot of lines where $this->assert* has been changed to self::assert* (technically, the correct way to call the asserts) however this wasn't done in this PR. My guess is that we're missing a PR somewhere that did all the changes from $this-> to self:: and that finding and merging that one first might reduce the number of conflicts in this PR or at the very least make it possible to see what's going on in the diffs. 

It is of cause possible that that PR will also have a huge number of conflicts, so it might be prudent to redo the work to change $this-> to self:: as it can be automated by php cs fixer.

Alternatively, if it's the project's choice to use $this-> to call the asserts php cs fixer could be run it in the opposite direction against the changes in this PR to make it closer to the current codebase before a merge is attempted.

**Comment by greg0ire** [comment]
I think redoing the change would be best, because you would not catch all occurrences anyway. We use phpcs in this project, maybe we can find a rule to add to our coding standard?

**Comment by carnage** [comment]
This is the rule class for php cs fixer https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/src/Fixer/PhpUnit/PhpUnitTestCaseStaticMethodCallsFixer.php but I can't find an equivalent in php codesniffer. We'd also need to decide which style to support (options being Assert::assertX, static::assertX, self::assertX and $this->assertX with the latter two being the usual choices)

**Comment by greg0ire** [comment]
🤔 I wonder why it is a phpunit-specific rule. My personal preference would be to run it once and make a PR, and pick `self`, because it won't conflict but because if we already merged that in the old-prototype-3.x, then there must be some consensus on using that.

**Comment by carnage** [comment]
In case anyone else comes across this ticket as a TODO, I have started work on it; however there are still ~60 conflicted files to go through so it's going to take a while :)

**Pull request merged**